pkg/agent/openflow/pipeline.go

```diff
@@ -2331,13 +2336,18 @@ func (f *featureEgress) snatRuleFlow(ofPort uint32, snatIP net.IP, snatMark uint
 	ipProtocol := getIPProtocol(snatIP)
 	if snatMark != 0 {
 		// Local SNAT IP.
-		return EgressMarkTable.ofTable.BuildFlow(priorityNormal).
+		fb := EgressMarkTable.ofTable.BuildFlow(priorityNormal).
 			Cookie(cookieID).
 			MatchProtocol(ipProtocol).
 			MatchCTStateNew(true).
 			MatchCTStateTrk(true).
-			MatchInPort(ofPort).
-			Action().LoadPktMarkRange(snatMark, snatPktMarkRange).
+			MatchInPort(ofPort)
+		if f.ovsMetersAreSupported {
+			if _, ok := f.cachedMeter.Load(snatMark); ok {
+				fb = fb.Action().Meter(snatMark)
+			}
+		}
```

Comment:
Now I get why all flows need update when bandwidth config changes. Could we have a separate table to enforce egress qos based on mark? This could decouple snat flow and qos flow and some code in egress controller. Currently it's a bit obscure to understand some code and order requirement in egress controller until digging into how the openflow is designed in this file.

Reply:
Sounds good. Changed.